### PR TITLE
Fix (#138): Topic train fragment Low-fi UI (DO NOT MERGE/REVIEW)

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

This PR contains low-fi implementation of Topic-Train-Fragment - UI

I have introduced `SkillInterface` and `SkillSelectionAdapter`. 
`SkillSelectionAdapter` is a custom RecyclerView Adapter which shows a list of skills along with checkboxes for selection from learner.
`SkillInterface` is responsible for transferring the selected skill item from `SkillSelectionAdapter` to `TopicTrainFragment`.

Mock: https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/f403add1-079a-4202-9975-efe04c076290/Home-Page-1-Start-Learning-Pre-Scroll-6

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is **assigned** to an appropriate reviewer.

### CODE REVIEW

**Comment by rt4914** [comment]
@veena14cs and @nikitamarysolomanpvt PTAL, so that we can transfer it to Ben for final review once your guys are done.

Also, @veena14cs please check it thoroughly because I has used a similar Adapter for recyclerview from your PR #196

File: app/src/main/java/org/oppia/app/topic/train/SkillInterface.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.oppia.app.topic.train
+
+/** Interface to update the selectedSkillList in [TopicTrainFragmentPresenter]. */
+interface SkillInterface {
+  /** This skill will get added to selectedSkillList in [TopicTrainFragmentPresenter] */
+  fun skillSelected(skill: String)
+  /** This skill will get removed from selectedSkillList in [TopicTrainFragmentPresenter] */
+  fun skillUnselected(skill: String)
+}
```
Comment (nikitamarysolomanpvt) [inline app/src/main/java/org/oppia/app/topic/train/SkillInterface.kt:9]: Eof

File: app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.topic_train_skill_view.view.*
+import org.oppia.app.R
+import org.oppia.app.databinding.TopicTrainSkillViewBinding
+
+/** Adapter to bind skills to [RecyclerView] inside [TopicTrainFragment]. **/
+class SkillSelectionAdapter(
+  private val skillList: List<String>,
+  private val skillInterface: SkillInterface
+) :
```
Comment (nikitamarysolomanpvt) [inline app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt:17]: Parentheses can be beside the parameters is't? or can avoid using separate line for  superclass immediately after colon

File: app/build.gradle [hunk 3]
```diff
@@ -61,6 +61,7 @@ dependencies {
       'androidx.constraintlayout:constraintlayout:1.1.3',
       'androidx.core:core-ktx:1.0.2',
       'androidx.lifecycle:lifecycle-runtime-ktx:2.2.0-alpha03',
+      'androidx.recyclerview:recyclerview:1.0.0',
       'com.google.dagger:dagger:2.24',
       "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version",
```
Comment (nikitamarysolomanpvt) [inline app/build.gradle:66]: Please change double quote to single quotes

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -12,4 +12,7 @@
   <string name="cellular_data_alert_dialog_title">Currently on Cellular Data</string>
   <string name="cellular_data_alert_dialog_description">Streaming audio may use a lot of cellular data.</string>
   <string name="cellular_data_alert_dialog_checkbox">Don\'t show this message again</string>
+  <string name="topic_train_master_these_skills">Master These Skills</string>
+  <string name="topic_train_skills_description">Select or unselect skills that you want to practice</string>
```
Comment (veena14cs) [inline app/src/main/res/values/strings.xml:16]: you can add full stop to the long messages

**Review by veena14cs — APPROVED** [review]
Generally LGTM

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (rt4914) [reply app/src/main/res/values/strings.xml:16]: The mock does not contain any `.` thats why, i kept it as it is.

File: app/build.gradle (hunk 3 above)
Comment (rt4914) [reply app/build.gradle:66]: Actually, I am not making changes to code which someone else has written.

File: app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt:17]: Done in PR #202

File: app/src/main/java/org/oppia/app/topic/train/SkillInterface.kt (hunk 1 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/train/SkillInterface.kt:9]: Done in PR #202

**Comment by rt4914** [comment]
This PR has been divided into 4 separate PRs mainly #200 #202 #203 #204

File: app/build.gradle (hunk 3 above)
Comment (nikitamarysolomanpvt) [reply app/build.gradle:66]: Last time in our discussions you suggested to fix issues if it comes in PR

**Pull request closed without merging**